src/main/java/com/sap/sgs/phosphor/fosstars/data/github/PackageManagement.java

```diff
@@ -0,0 +1,232 @@
+import static com.sap.sgs.phosphor.fosstars.model.value.PackageManager.YARN;
+
+import com.sap.sgs.phosphor.fosstars.model.Value;
+import com.sap.sgs.phosphor.fosstars.model.ValueSet;
+import com.sap.sgs.phosphor.fosstars.model.value.Language;
+import com.sap.sgs.phosphor.fosstars.model.value.Languages;
+import com.sap.sgs.phosphor.fosstars.model.value.PackageManager;
+import com.sap.sgs.phosphor.fosstars.model.value.PackageManagers;
+import com.sap.sgs.phosphor.fosstars.model.value.ValueHashSet;
+import java.io.IOException;
+import java.util.EnumMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Predicate;
+import org.kohsuke.github.GHContent;
+import org.kohsuke.github.GHRepository;
+import org.kohsuke.github.GitHub;
+
+/**
+ * This data provider returns package managers which are used in a project.
+ */
+public class PackageManagement extends AbstractGitHubDataProvider {
+
+  /**
+   * A minimal number of characters in a config to consider it valid.
+   */
+  private static final int ACCEPTABLE_CONFIG_SIZE = 100;
+
+  /**
+   * Maps a programming language to its package manager.
+   */
+  private static final Map<Language, PackageManagers> KNOWN_PACKAGE_MANAGERS
+      = new EnumMap<>(Language.class);
+
+  static {
+    register(JAVA, MAVEN, GRADLE);
+    register(SCALA, MAVEN);
+    register(JAVASCRIPT, NPM, YARN);
+    register(C_SHARP, DOTNET);
+    register(F_SHARP, DOTNET);
+    register(VISUALBASIC, DOTNET);
+    register(PYTHON, PIP);
+    register(RUBY, RUBYGEMS);
+    register(PHP, COMPOSER);
+  }
+
+  /**
+   * Maps a package manager to a list of its possible config files.
+   *
+   * @see <a href="https://help.github.com/en/github/visualizing-repository-data-with-graphs/listing-the-packages-that-a-repository-depends-on">
+   *      Listing the packages that a repository depends on</a>
+   */
+  private static final Map<PackageManager, Predicate<String>[]> CONFIG_FILES_PATTERNS
+      = new EnumMap<>(PackageManager.class);
+
+  static {
+    register(MAVEN, "pom.xml"::equals);
+    register(GRADLE, "build.gradle"::equals);
+    register(NPM, "package-lock.json"::equals, "package.json"::equals);
+    register(YARN, "yarn.lock"::equals, "package.json"::equals);
+    register(DOTNET, ".csproj"::equals, ".vbproj"::equals, ".nuspec"::equals,
+        ".vcxproj"::equals, ".fsproj"::equals, "packages.config"::equals);
+    register(RUBYGEMS, "Gemfile.lock"::equals, "Gemfile"::equals, ".gemspec"::endsWith);
+    register(COMPOSER, "composer.json"::equals, "composer.lock"::equals);
+  }
+
+  /**
+   * Adds a number of known package managers for a language.
+   *
+   * @param language The language.
+   * @param list The package managers.
+   */
+  private static void register(Language language, PackageManager... list) {
+    if (KNOWN_PACKAGE_MANAGERS.containsKey(language)) {
+      throw new IllegalArgumentException(String.format("%s is already there!", language));
```

Comment:
Hmm instead of throwing the exception straight away, how about checking if it's a new packagemanager not associated with the language yet?

Reply:
This check is done only once when the class is loaded. I would prefer to keep it as simple as possible and allow specifying a list of package managers for a language only once.